src/TextAnnotator.jsx

```diff
@@ -340,11 +342,34 @@ export default class TextAnnotator extends Component {
     this.setState({ readOnly });
   }
 
+  get widgets() {
+    return this.state.widgets;
+  }
+
+  set widgets(widgets) {
+    this.setState({ widgets });
+  }
+
+  get disableEditor() {
+    return this.state.editorDisabled;
+  }
+
+  set disableEditor(disabled) {
+    this.setState({editorDisabled: disabled}, () => {
+      // En- or disable Esc key listener
+      if (disabled)
+        document.addEventListener('keyup', this.handleEscape);
+      else
+        document.removeEventListener('keyup', this.handleEscape);
+    });
+  }
+
   render() {
   	// The editor should open under normal conditions - annotation was selected, no headless mode
     const open = (this.state.selectedAnnotation || this.state.selectedRelation) && !this.state.editorDisabled;
 
-    const readOnly = this.state.readOnly || this.state.selectedAnnotation?.readOnly
+    const readOnly = this.state.readOnly || this.state.selectedAnnotation?.readOnly;
+    const widgets = this.state.widgets || this.state.selectedAnnotation?.widgets;
```

Comment:
This line isn't necessary. You can just use `{this.state.widgets}` directly in the JSX. The reason why this line exists for the `readOnly` property is because `readOnly` can be a global setting (everything on the image is read-only), but also a local setting specific to one annotation (the image might be annotate-able, but one annotation is read-only, maybe because your environment has user permissions, and it's someone else's annotation).

I.e. the fix is simply to remove that line and then use `{this.state.widgets}` below in the markup. But don't worry. I can make that fix after merging.